cds/modules/webhooks/receivers.py

```diff
@@ -26,123 +26,33 @@
-        raise NotImplemented()
-
-
-class CeleryTaskReceiver(TaskReceiver):
-    """Base class for Celery-based TaskReceivers.
-    Implementing this class requires the definition of a single Celery task.
-    .. note::
-        Arguments of the task must match the ones of the task provided exactly.
-    """
-
-    @property
-    def celery_task(self):
-        """Celery task to be executed by this receiver."""
-        raise NotImplementedError
-
-    def new_task(self, task_id, kwargs):
-        """Start asynchronous execution of Celery task."""
-        return self.celery_task.apply_async(task_id=task_id, kwargs=kwargs).id
-
-    def get_status(self, task_id):
-        """Retrieve status of current task from the Celery backend."""
-        result = AsyncResult(task_id)
-        return result.state, result.info if result.state == 'PROGRESS' else {}
-
-    def cancel_task(self, task_id):
-        """Cancel execution of the Celery task."""
-        AsyncResult(task_id).revoke(terminate=True)
-        return 'Revoked task'
-
-
-class CeleryChainTaskReceiver(TaskReceiver):
-    """TaskReceiver specialized for Celery."""
-
-    # List of tuples of the form (celery_task, argument identifiers)
     @property
-    def celery_tasks(self):
+    def workflow(self):
         raise NotImplemented()
 
-    def new_task(self, task_id, kwargs):
+    def run(self, event):
         """Construct Celery canvas.
 
         This is achieved by chaining sequential tasks and grouping
         concurrent ones.
         """
-        task_list = []
-        parent_kw = {'parent_id': task_id}
-        for task_definition in self.celery_tasks:
-            if isinstance(task_definition, tuple):
-                task, task_kw = task_definition
-                kw = {k: kwargs[k] for k in kwargs if k in task_kw}
-                kw.update(parent_kw)
-                task_list.append(task.subtask(kwargs=kw))
-            elif isinstance(task_definition, list):
-                subtasks = []
-                for task, task_kw in task_definition:
-                    kw = {k: kwargs[k] for k in kwargs if k in task_kw}
-                    kw.update(parent_kw)
-                    subtasks.append(task.subtask(kwargs=kw))
-                task_list.append(group(*subtasks))
-        return chain(*task_list, task_id=task_id)().id
-
-    def get_status(self, task_id):
-        result = AsyncResult(task_id)
-        if result.state in ['PROGRESS', 'EXCEPTION']:
-            return result.state, result.info
-        else:
-            return {}
-
-    def cancel_task(self, task_id):
-        AsyncResult(task_id).revoke(terminate=True)
-        return 'Revoked task'
-
-
-class AVWorkflow(CeleryChainTaskReceiver):
-    """Composite CeleryChainTaskReceiver for the AV workflow."""
-
-    celery_tasks = [
+        # FIXME Be able to set base class for `process_event` task
+        chain_orchestrator.apply_async(
+            (self.workflow, ), kwargs=event.payload, task_id=event.id)
```

Comment:
(a) will make it impossible to get the Celery status from the REST API (https://github.com/inveniosoftware/invenio-webhooks/pull/37)
(b) That works, but still there is a redundant task, which calls the orchestrator task. IMHO it would be cleaner and more flexible to allow different bases for `process_events`.